tests/countries/test_cayman_islands.py

```diff
@@ -0,0 +1,255 @@
+        )
+        self.assertHolidayName(name, range(2024, 2050))
+        self.assertNoHolidayName(name, range(2006, 2024))
+
+    def test_discovery_day(self):
+        name = "Discovery Day"
+        self.assertHolidayName(
+            name,
+            "2020-05-18",
+            "2021-05-17",
+            "2022-05-16",
+            "2023-05-15",
+            "2024-05-20",
+            "2025-05-19",
+        )
+        self.assertHolidayName(name, range(2006, 2050))
+
+    def test_queens_birthday(self):
+        name = "Queen's Birthday"
+        self.assertHolidayName(
+            name,
+            "2018-06-11",
+            "2019-06-10",
+            "2020-06-15",
+            "2021-06-14",
+            "2022-06-06",
+        )
+        self.assertHolidayName(name, range(2006, 2023))
+        self.assertNoHolidayName(name, range(2023, 2050))
+
+    def test_kings_birthday(self):
+        name = "King's Birthday"
+        self.assertHolidayName(
+            name,
+            "2023-06-19",
+            "2024-06-17",
+            "2025-06-23",
+        )
+        self.assertNoHolidayName(name, range(2006, 2023))
+
+    def test_constitution_day(self):
+        name = "Constitution Day"
+        self.assertHolidayName(
+            name,
+            "2020-07-06",
+            "2021-07-05",
+            "2022-07-04",
+            "2023-07-03",
+            "2024-07-01",
+            "2025-07-07",
+        )
+        self.assertHolidayName(name, range(2006, 2050))
+
+    def test_remembrance_day(self):
+        name = "Remembrance Day"
+        self.assertHolidayName(
+            name,
+            "2020-11-09",
+            "2021-11-08",
+            "2022-11-14",
+            "2023-11-13",
+            "2024-11-11",
+            "2025-11-10",
+        )
+        self.assertHolidayName(name, range(2006, 2050))
+
+    def test_christmas_day(self):
+        name = "Christmas Day"
+        self.assertHolidayName(name, (f"{year}-12-25" for year in range(2006, 2050)))
+        obs_dt = (
+            "2010-12-27",
+            "2011-12-27",
+            "2016-12-27",
+            "2021-12-27",
+            "2022-12-27",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_boxing_day(self):
+        name = "Boxing Day"
+        self.assertHolidayName(name, (f"{year}-12-26" for year in range(2006, 2050)))
+        obs_dt = (
+            "2010-12-28",
+            "2015-12-28",
+            "2020-12-28",
+            "2021-12-28",
+        )
+        self.assertHolidayName(f"{name} (observed)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_referendum_day(self):
+        name = "Referendum Day"
+        self.assertHolidayName(name, "2012-07-18", "2019-12-19")
+        self.assertNoHolidayName(name, range(2006, 2012))
+        self.assertNoHolidayName(name, range(2013, 2019))
+        self.assertNoHolidayName(name, range(2020, 2050))
```

Comment:
IMO it might be better to lump General Election Day and Referendum Day testcases with `test_special_holidays` for simpler maintenance

If you wishes to test the holiday names as well for these dates, see https://github.com/vacanza/holidays/blob/c3e0a2ce1ba0b3b0f00d7c1da98a7189e200bcab/tests/countries/test_anguilla.py#L28-L36